includes/templates/class-amp-post-template.php

```diff
@@ -302,15 +302,19 @@ private function build_post_data() {
 	 * @return int Post publish UTC timestamp.
 	 */
 	private function build_post_publish_timestamp() {
-		$format    = 'U';
-		$timestamp = (int) get_post_time( $format, true, $this->post, true );
+		if ( empty( $this->post->post_date_gmt ) || '0000-00-00 00:00:00' === $this->post->post_date_gmt ) {
+			$timestamp = time();
+		} else {
+			$format    = 'U';
+			$timestamp = (int) get_post_time( $format, true, $this->post, true );
 
-		/** This filter is documented in wp-includes/general-template.php. */
-		$filtered_timestamp = apply_filters( 'get_the_date', $timestamp, $format, $this->post );
```

Comment:
Should the `$timestamp` not still get filtered when it is `time()`?

Reply:
Addressed in https://github.com/ampproject/amp-wp/pull/5450/commits/db113382378c66b523c099ae182a2a12ab5b5bb4.